README.md
# G42-200 Starliner
The G42-200 is a state-of-the-art aircraft capable of accelerating out of the atmosphere and into low earth orbit - Categorized as a spaceplane, 
the G42 features airliner-like handling characteristics which makes it optimal for low-cost space travel and cargo lifting based on conventional or mildly adapted airfields 
at a fraction of the costs usually associated with space operations.
 
Development thread: [Updated G42-200 Starliner](https://www.orbiter-forum.com/showthread.php?t=41302)
 
## General features
- Highly detailed DVC with oodles of clickable switches, 6 functional EICAS displays and 6 MFD's.
- Realistic flight dynamics and engine model, featuring simulated fuel pumps, startup/shutdown, wing positions, flight controls and more.
- A wealth of animated moving parts on both the VC and the exterior model.
- Custom sounds including proper turbines ramp-up/down effects.
- A large bay that can carry up to 8 UCSO cargoes.
- Improved model and textures since the last release.
- Full D3D9 graphics client compatibility.

CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## WIP 307 - 2021-03-04
### Added
- New scenarios and images by Matias Saibene.
- Burner engage knob.
- Main, ASF, and oxidizer tanks full, gear up, gear up and locked, gear down, and gear down and locked announcements.
- Gear is up warning if below 300 meters and descending.
- APU offline warning if tried to manipulate a moving part and the APU is off.
- Antislip AP on and off sounds.
- Docking port can't be used if the payload bay doors are closed.
- Engines contrail.
- Gear indicator on the HUD.
- Center of gravity shifting.
- ApT and PeT on the orbit HUD.
- Shift + S shortcut to start or stop the engines.
- Percent sign in the APU quantity to make it more clear.
- UCSO cargoes can now be packed and unpacked.
### Changed
- Dynamic XRSound is used instead of Orbiter Sound SDK.
- Hydraulic pressure is now required to operate the wheel brakes.
- Moving parts "saving" logic.
- Renamed landing gears to landing gear.
- Sound implementation for moving parts.
- Repository structure.
- UCSO operation way. Cargo mode have to be activated (Shift + C) which will change the HUD display.
- User manual.
- Engine positions are now correct on X and Z axis (Y axis correction results in undesired flight characteristics).
- Engines are now fully separated, they no longer act as a one unit.
- Touchdown points so landing gear is no longer below the ground.
- Keyboard shortcuts to call switches instead of doing work.
- APU sounds are separated.
### Fixed
- Working stuff sound not played when loading an existing scenario.
- Oxidizer pumps not turning off if the fuel line is closed.
- Messing OMS and RCS fuel pump sounds.
- Extend/Retract switches not switching back to original position if moved to the current position.
- Hydraulic pressure not correctly aligned in the EICAS.
- Wing position changes (although not completely) if APU isn't on.
- Oxidizer flow showing 0.
- VC seat position not saved.
- Fuel pressures not shown when the engine is not working.
- Reduced the tick sound in the afterburner sound.
- Wing position not correctly read from scenario.

## WIP 306 - 2020-02-28
### Added
- Header file generation to the 3DS Max Orbiter mesh tool.
- Landing, taxi, and docking light.
- ADI ball.
- 2 Hydraulic systems.
- Gear emergency release system.
- Fuel lines.
- Undocking handle.
- HUD color and brightness buttons.
- Anti-slip autopilot.
- Fuel dump system.
- Bay door switch.
- Docking supply system.
- Warning system, with V1 and rotate callouts.
- RCS doors logic.
- APU Pack B.
### Changed
- 3DS Max tools are now utilities instead of macro scripts.
- RCS and OMS are now separated systems.
- Ramcaster engine coefficient.
- Landing gears damping coefficient.
### Removed
- Useless and inoperative switches.

## WIP 305 - 2020-02-06
### Added
- UCSO support.
- Missing cockpit sounds.
- Parking brake with handle in the cockpit.
- Ramcaster engine handle is clickable in the cockpit.
- Autopilot and HUD mode buttons.
- RCS and ACS knobs.
- APU EICAS screen.
- Docking RCS mode.
- RCS and OMS engines fuel pumps.
- APU must be on to engage ACS.
### Fixed
- Touchdown points.
- Vessel size is set 1 meter.
- Scenario load bug.
- Nosewheel steering.
- Wheels braking.
- HUD data.
- Landing gear sound not played when extending the gear.
- APU cycle switch out of place.
- HUD is too dim sometimes.
- Wing position indicator on ACS EICAS screen reversing the actual wing position.
### Changed
- Source code cleanup. No more compiler warnings.
- Upgraded Orbiter Sound SDK to 4.0.
- EICAS engine displays are now for both engines. Before, both engine displays are only showing the left engine values.
- Textures folder is now G422 instead of DGa2.
- Vertical lift coefficient.
- Upgraded to WIP 304 mesh (new EICAS screens).
- The checklist to match the current version.
- Ramcaster engine coefficient.
- Revised EICAS displays.
### Removed
- Useless files, libraries and textures.
- APU Pack B.

Sources/G422/Header/G422_MDL_DVC.h
#ifndef G422_MDL_DVC
#define G422_MDL_DVC

#define TEX_MFDKEYS 5
#define TEX_EICAS   7

#define MGID_Shape001 0
#define MGID_Line006 1
#define MGID_Plane001 2
#define MGID_Box007 3
#define MGID_Object023 4
#define MGID_Object025 5
#define MGID_Object026 6
#define MGID_Object027 7
#define MGID_Object028 8
#define MGID_Object029 9
#define MGID_Plane002 10
#define MGID_Box008 11
#define MGID_Object048 12
#define MGID_MFD_6 13
#define MGID_MFD_4 14
#define MGID_MFD_KEYS 15
#define MGID_Object057 16
#define MGID_Object058 17
#define MGID_Object059 18
#define MGID_Box031 19
#define MGID_Object065 20
#define MGID_Object148 21
#define MGID_Object149 22
#define MGID_Object150 23
#define MGID_Box032 24
#define MGID_Object326 25
#define MGID_Cylinder004 26
#define MGID_Object328 27
#define MGID_Object329 28
#define MGID_Object330 29
#define MGID_RAMX_THR_GRIP 30
#define MGID_RAMX_THR_STICK 31
#define MGID_THR_RAMX_DETENT 32
#define MGID_RAMX_THR_FLAP 33
#define MGID_EMC_TRIGGER_COVER 34
#define MGID_WING_POS_LEVER 35
#define MGID_Cylinder006 36
#define MGID_THROTTLE_UNIT 37
#define MGID_Object336 38
#define MGID_Object337 39
#define MGID_Object338 40
#define MGID_Object339 41
#define MGID_PRK_BRK_HNDL 42
#define MGID_KB2_COORD_REF 43
#define MGID_Object342 44
#define MGID_KB3_ACS_MODE 45
#define MGID_KB3_RCS_MODE 46
#define MGID_Object345 47
#define MGID_STICK_BASE 48
#define MGID_CTRL_STICK 49
#define MGID_Object350 50
#define MGID_Object351 51
#define MGID_Object352 52
#define MGID_MFD_1 53
#define MGID_MFD_2 54
#define MGID_MFD_3 55
#define MGID_Object355 56
#define MGID_Object357 57
#define MGID_VISOR_FRONT_OUT 58
#define MGID_VISOR_FRONT_IN 59
#define MGID_Object361 60
#define MGID_SW3_GEAR 61
#define MGID_SW3_CANARD 62
#define MGID_SW3_VISOR 63
#define MGID_SW3_RCS 64
#define MGID_Object363 65
#define MGID_Object364 66
#define MGID_Object365 67
#define MGID_Object1226 68
#define MGID_Object1228 69
#define MGID_Object1230 70
#define MGID_Object1231 71
#define MGID_Object1232 72
#define MGID_Object1233 73
#define MGID_SW3_EMAIN_MODE 74
#define MGID_SW3_EMAIN_REHEAT 75
#define MGID_SW2_LGT_NAV 76
#define MGID_SW2_LGT_STB 77
#define MGID_SW2_LGT_BCN 78
#define MGID_SW2_LGT_LGR 79
#define MGID_SW2_LGT_LND 80
#define MGID_SW2_LGT_TAX 81
#define MGID_SW2_LGT_DCK 82
#define MGID_SW2_LGT_BAY 83
#define MGID_SW2_LGT_PNL 84
#define MGID_SW2_LGT_FLD 85
#define MGID_SW2_LGT_CSL 86
#define MGID_SW2_LGT_DOM 87
#define MGID_SW2_LGT_AMB 88
#define MGID_SW2_LGT_BRT 89
#define MGID_SW3_RAMX_DOOR 90
#define MGID_SW3_RAMX_IGN 91
#define MGID_SW3_RAMX_COOL 92
#define MGID_SW3_RAMX_MODE 93
#define MGID_SW3_EMAIN_IGN 94
#define MGID_SW3_EMAIN_BRNRIGN 95
#define MGID_SW3_STBYIGN_RCS 96
#define MGID_SW3_STBYIGN_OMS 97
#define MGID_SW2_MSFEED_C 98
#define MGID_SW2_MSFEED_L 99
#define MGID_SW2_MSFEED_R 100
#define MGID_SW2_SYSFEED_APU 101
#define MGID_SW2_SYSFEED_OMS 102
#define MGID_SW2_SYSFEED_RCS 103
#define MGID_SW3_OXYFEED_L 104
#define MGID_SW3_OXYFEED_R 105
#define MGID_SW2_THR_AUTH 106
#define MGID_SW2_BRKCOOL 107
#define MGID_SW3_APU_CYCLE 108
#define MGID_SW3_APU_COOL 109
#define MGID_SW3_APU_EXHT 110
#define MGID_SW2_APU_RAMX 111
#define MGID_SW3_APU_PACK_A 112
#define MGID_SW3_APU_PACK_B 113
#define MGID_SW3_APU_AC 114
#define MGID_SW3_APU_BLEED 115
#define MGID_SW3_APU_HYD 116
#define MGID_EICAS_C2 117
#define MGID_FLRES_DSP 118
#define MGID_MFD_5 119
#define MGID_Object1237 120
#define MGID_STICK_BOOT_1 121
#define MGID_STICK_BOOT_2 122
#define MGID_Object1344 123
#define MGID_SW2_EXTANK_L 124
#define MGID_SW2_EXTANK_B 125
#define MGID_SW2_EXTANK_R 126
#define MGID_Object1345 127
#define MGID_Object1347 128
#define MGID_Object1348 129
#define MGID_Object1349 130
#define MGID_Object1350 131
#define MGID_Object1351 132
#define MGID_Object1352 133
#define MGID_Object1353 134
#define MGID_Object1354 135
#define MGID_Object1355 136
#define MGID_Object1356 137
#define MGID_Object1357 138
#define MGID_Object1358 139
#define MGID_Object1360 140
#define MGID_Object1361 141
#define MGID_Object1362 142
#define MGID_Object1364 143
#define MGID_Object1365 144
#define MGID_Object1366 145
#define MGID_Object1367 146
#define MGID_Object1368 147
#define MGID_Object1370 148
#define MGID_Object1371 149
#define MGID_Object1372 150
#define MGID_Object1375 151
#define MGID_Object1376 152
#define MGID_Object1377 153
#define MGID_Object1378 154
#define MGID_ADI_RING 155
#define MGID_ADI_BALL 156
#define MGID_Object1379 157
#define MGID_Object1380 158
#define MGID_Cylinder008 159
#define MGID_SW3_BAY_OPENCLSE 160
#define MGID_SW3_BAY_LATCH 161
#define MGID_SW3_RADIATOR 162
#define MGID_Object1381 163
#define MGID_Object1382 164
#define MGID_VC_SWITCHES_OVHD_DULL001 165
#define MGID_VC_SWITCHES_OVHD_DULL002 166
#define MGID_VC_SWITCHES_OVHD_DULL003 167
#define MGID_SW2_DCKSPL_FUEL 168
#define MGID_SW2_DCKSPL_OXY 169
#define MGID_SW2_DCKSPL_ASF 170
#define MGID_VC_SWITCHES_OVHD_DULL007 171
#define MGID_VC_SWITCHES_OVHD_DULL008 172
#define MGID_VC_SWITCHES_OVHD_DULL009 173
#define MGID_VC_SWITCHES_OVHD_DULL010 174
#define MGID_VC_SWITCHES_OVHD_DULL011 175
#define MGID_VC_SWITCHES_OVHD_DULL012 176
#define MGID_VC_SWITCHES_OVHD_DULL013 177
#define MGID_VC_SWITCHES_OVHD_DULL014 178
#define MGID_VC_SWITCHES_OVHD_DULL015 179
#define MGID_VC_SWITCHES_OVHD_DULL016 180
#define MGID_VC_SWITCHES_OVHD_DULL017 181
#define MGID_VC_SWITCHES_OVHD_DULL018 182
#define MGID_VC_SWITCHES_OVHD_DULL019 183
#define MGID_VC_SWITCHES_OVHD_DULL020 184
#define MGID_VC_SWITCHES_OVHD_DULL021 185
#define MGID_VC_SWITCHES_OVHD_DULL022 186
#define MGID_VC_SWITCHES_OVHD_DULL023 187
#define MGID_VC_SWITCHES_OVHD_DULL024 188
#define MGID_VC_SWITCHES_OVHD_DULL025 189
#define MGID_VC_SWITCHES_OVHD_DULL026 190
#define MGID_VC_SWITCHES_OVHD_DULL027 191
#define MGID_VC_SWITCHES_OVHD_DULL028 192
#define MGID_VC_SWITCHES_OVHD_DULL029 193
#define MGID_VC_SWITCHES_OVHD_DULL030 194
#define MGID_VC_SWITCHES_OVHD_DULL031 195
#define MGID_VC_SWITCHES_OVHD_DULL032 196
#define MGID_VC_SWITCHES_OVHD_DULL033 197
#define MGID_VC_SWITCHES_OVHD_DULL034 198
#define MGID_VC_SWITCHES_OVHD_DULL035 199
#define MGID_VC_SWITCHES_OVHD_DULL036 200
#define MGID_VC_SWITCHES_OVHD_DULL037 201
#define MGID_VC_SWITCHES_OVHD_DULL038 202
#define MGID_VC_SWITCHES_OVHD_DULL039 203
#define MGID_VC_SWITCHES_OVHD_DULL040 204
#define MGID_VC_SWITCHES_OVHD_DULL041 205
#define MGID_VC_SWITCHES_OVHD_DULL042 206
#define MGID_VC_SWITCHES_OVHD_DULL043 207
#define MGID_VC_SWITCHES_OVHD_DULL044 208
#define MGID_VC_SWITCHES_OVHD_DULL045 209
#define MGID_VC_SWITCHES_OVHD_DULL046 210
#define MGID_VC_SWITCHES_OVHD_DULL047 211
#define MGID_VC_SWITCHES_OVHD_DULL048 212
#define MGID_VC_SWITCHES_OVHD_DULL049 213
#define MGID_VC_SWITCHES_OVHD_DULL050 214
#define MGID_VC_SWITCHES_OVHD_DULL051 215
#define MGID_VC_SWITCHES_OVHD_DULL052 216
#define MGID_VC_SWITCHES_OVHD_DULL053 217
#define MGID_VC_SWITCHES_OVHD_DULL054 218
#define MGID_VC_SWITCHES_OVHD_DULL055 219
#define MGID_VC_SWITCHES_OVHD_DULL056 220
#define MGID_VC_SWITCHES_OVHD_DULL057 221
#define MGID_VC_SWITCHES_OVHD_DULL058 222
#define MGID_VC_SWITCHES_OVHD_DULL059 223
#define MGID_VC_SWITCHES_OVHD_DULL060 224
#define MGID_VC_SWITCHES_OVHD_DULL061 225
#define MGID_VC_SWITCHES_OVHD_DULL062 226
#define MGID_VC_SWITCHES_OVHD_DULL063 227
#define MGID_SW2_ASF_DUMP 228
#define MGID_SW2_OXY_DUMP 229
#define MGID_SW2_RIGHT_DUMP 230
#define MGID_SW2_FWD_DUMP 231
#define MGID_SW2_AFT_DUMP 232
#define MGID_SW2_LEFT_DUMP 233
#define MGID_VC_SWITCHES_OVHD_DULL070 234
#define MGID_VC_SWITCHES_OVHD_DULL071 235
#define MGID_VC_SWITCHES_OVHD_DULL075 236
#define MGID_OVHD_FIRE_SWITCHES001 237
#define MGID_OVHD_FIRE_SWITCHES002 238
#define MGID_OVHD_FIRE_SWITCHES003 239
#define MGID_OVHD_FIRE_SWITCHES004 240
#define MGID_OVHD_FIRE_SWITCHES005 241
#define MGID_OVHD_FIRE_SWITCHES006 242
#define MGID_OVHD_FIRE_SWITCHES007 243
#define MGID_OVHD_FIRE_SWITCHES008 244
#define MGID_OVHD_FIRE_SWITCHES009 245
#define MGID_OVHD_FIRE_SWITCHES010 246
#define MGID_SW2_MASTER_DUMP 247
#define MGID_SWITCHES_DULL001 248
#define MGID_SWITCHES_DULL002 249
#define MGID_SWITCHES_DULL003 250
#define MGID_SWITCHES_DULL004 251
#define MGID_SWITCHES_DULL005 252
#define MGID_SWITCHES_DULL006 253
#define MGID_Object1388 254
#define MGID_VC_SWITCHES_OVHD_DULL076 255
#define MGID_VC_SWITCHES_OVHD_DULL077 256
#define MGID_VC_SWITCHES_OVHD_DULL078 257
#define MGID_VC_SWITCHES_OVHD_DULL079 258
#define MGID_VC_SWITCHES_OVHD_DULL080 259
#define MGID_VC_SWITCHES_OVHD_DULL081 260
#define MGID_VC_SWITCHES_OVHD_DULL082 261
#define MGID_VC_SWITCHES_OVHD_DULL083 262
#define MGID_UNDOCK_HNDL 263
#define MGID_SW3_DARTS 264
#define MGID_Object1391 265
#define MGID_Object1392 266
#define MGID_GEAR_EMER_HNDL 267
#define MGID_GND_STEER_TILLER 268
#define MGID_VC_SWITCHES_OVHD_DULL084 269
#define MGID_VC_SWITCHES_OVHD_DULL085 270
#define MGID_VC_SWITCHES_OVHD_DULL086 271
#define MGID_VC_SWITCHES_OVHD_DULL087 272
#define MGID_Object1394 273
#define MGID_VC_SWITCHES_OVHD_DULL088 274
#define MGID_VC_SWITCHES_OVHD_DULL089 275
#define MGID_VC_SWITCHES_OVHD_DULL090 276
#define MGID_VC_SWITCHES_OVHD_DULL091 277
#define MGID_VC_SWITCHES_OVHD_DULL092 278
#define MGID_VC_SWITCHES_OVHD_DULL093 279
#define MGID_SW3_RAMX_IGN001 280
#define MGID_SW3_RAMX_DOOR001 281
#define MGID_SW3_RAMX_IGN002 282
#define MGID_VC_SWITCHES_OVHD_DULL094 283
#define MGID_VC_SWITCHES_OVHD_DULL095 284
#define MGID_VC_SWITCHES_OVHD_DULL096 285
#define MGID_VC_SWITCHES_OVHD_DULL097 286
#define MGID_VC_SWITCHES_OVHD_DULL098 287
#define MGID_VC_SWITCHES_OVHD_DULL099 288
#define MGID_VC_SWITCHES_OVHD_DULL100 289
#define MGID_VC_SWITCHES_OVHD_DULL101 290
#define MGID_VC_SWITCHES_OVHD_DULL102 291
#define MGID_VC_SWITCHES_OVHD_DULL103 292
#define MGID_VC_SWITCHES_OVHD_DULL104 293
#define MGID_VC_SWITCHES_OVHD_DULL105 294
#define MGID_VC_SWITCHES_OVHD_DULL106 295
#define MGID_VC_SWITCHES_OVHD_DULL107 296
#define MGID_SW3_ENGPNL_001 297
#define MGID_VC_SWITCHES_OVHD_DULL108 298
#define MGID_VC_SWITCHES_OVHD_DULL109 299
#define MGID_VC_SWITCHES_OVHD_DULL110 300
#define MGID_VC_SWITCHES_OVHD_DULL111 301
#define MGID_VC_SWITCHES_OVHD_DULL112 302
#define MGID_VC_SWITCHES_OVHD_DULL113 303
#define MGID_EICAS_C1 304
#define MGID_EICAS_P1 305
#define MGID_EICAS_P2 306
#define MGID_EICAS_E2 307
#define MGID_EICAS_E1 308
#define MGID_WINDSHIELD_THINGY 309
#define MGID_Object360 310
#define MGID_VISOR_AFT_IN 311
#define MGID_VISOR_WINDOW_FRAME 312
#define MGID_VISOR_MECH 313
#define MGID_Object1229 314
#define MGID_KB2_RENG_FEED 315
#define MGID_KB2_RAMX_FEED 316
#define MGID_KB2_LENG_FEED 317
#define MGID_KB2_RCS_FEED 318
#define MGID_KB2_OMS_FEED 319
#define MGID_KB2_APU_FEED 320
#define MGID_KB2_LOXY_FEED 321
#define MGID_KB2_ROXY_FEED 322
#define MGID_Object147 323
#define MGID_HUD 324

#endif